doc/proposals/sdk-integration-with-olm.md

```diff
@@ -0,0 +1,212 @@
+
+## Motivation
+
+OLM is an incredibly useful cluster management tool. There is currently no integration between SDK and OLM that encourages running an Operator with the latter.
+
+### Goals
+
+#### General
+
+* Operator developers can use `operator-sdk` to quickly deploy OLM on a given Kubernetes cluster
+* Operator developers can use `operator-sdk` to run their Operator under OLM
+* Operator developers can use `operator-sdk` to build a catalog/bundle containing their Operator for use with OLM
+
+#### Specific
+
+* `operator-sdk` creates a [bundle][bundle] from an Operator project to deploy with OLM
+* `operator-sdk` has a CLI interface to interact with OLM
+* `operator-sdk` installs a specific version of OLM onto Kubernetes cluster
+* `operator-sdk` uninstalls a specific version of OLM onto Kubernetes cluster
+* `operator-sdk` accepts a bundle and deploys that operator onto an OLM-enabled Kubernetes cluster
+* `operator-sdk` accepts a bundle and removes that operator onto an OLM-enabled Kubernetes cluster
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
+**TODO**
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+
+### Implementation Details/Notes/Constraints
+
+Initial PR: https://github.com/operator-framework/operator-sdk/pull/1912
+
+#### Use of operator-registry
+
+The SDK's approach to deployment should be as general and reliant on existing mechanisms as possible. To that end, [`operator-registry`][registry] should be used since it defines what a bundle contains and how one is structured. `operator-registry` libraries should be used to create and serve bundles, and interact with package manifests.
+
+The idea is to create a `Deployment` containing the latest `operator-registry` [image][registry-image] to initialize a bundle database and run a registry server serving that database using binaries contained in the image. The `Deployment` will contain volume mounts from a `ConfigMap` containing bundle files and a package manifest for an operator. Using manifest data in the `ConfigMap` volume source, the registry initializer can build a local database and serve that database through the `Service`. OLM-specific resources created by the SDK or supplied by a user, described below, will establish communication between this registry server and OLM.
```

Comment:
This part is not clear to me. Will we change what a bundle is currently?.

Should we not just generate the required files which in order to create which it calls a bundle the in the path `project/deploy/olm-catalog/<project-name>`?

So, in this way, I understand that:

- The CSV file and CRD's would be created (which is done currently)
- then, we need to create the pkg file as this [example](https://github.com/dev4devs-com/postgresql-operator/blob/master/deploy/olm-catalog/postgresql-operator/postgresql-operator.package.yaml0

It would be the bundle. 

- Also, we should follow the same standard/ impl of https://operatorhub.io/bundle
- And we could add here the [doc](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#bundle-format) which has the steps to create the bundle.